### PR TITLE
[10-10CG] Update pdf filler with new 1010-CG pdf

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- This work is behind a feature toggle (flipper): NO
- Added new version of 10-10CG form pdf. This pdf is used in two places. First, it is included in the payload as a signed pdf when the application is submitted. Second, it is available for download (without signatures) by the Veteran after they submit the application.
- Currently the 2021 version of the pdf is being filled out, which is missing the updated fields

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/86927

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
10-10CG 

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: spec/lib/pdf_fill/filler_spec.rb [hunk 1]
```diff
@@ -59,26 +59,24 @@
         factory: :income_and_assets_claim,
         use_vets_json_schema: true
       },
-      # Removing these test cases temporarily until we can resolve the fixture issues
```
Comment (coope93) [inline spec/lib/pdf_fill/filler_spec.rb:62]: These tests were previously commented out and are now being run.

File: lib/pdf_fill/forms/va1010cg.rb [hunk 2]
```diff
@@ -10,103 +10,127 @@ class Va1010cg < FormBase
       PDF_INPUT_LOCATIONS = OpenStruct.new(
```
Comment (coope93) [inline lib/pdf_fill/forms/va1010cg.rb:10]: These changes correspond to the form fields in the pdf version of the 10-10CG. The field names were updated to match the updated versions of the fields on the new form.

File: lib/pdf_fill/forms/va1010cg.rb [hunk 3]
```diff
@@ -119,18 +143,13 @@ class Va1010cg < FormBase
...
                 limit: 80,
-                question_num: 7,
+                question_num: 108,
                 question_text: 'VETERAN/SERVICEMEMBER > Address > Street'
               }
             },
             'gender' => {
               key: PDF_INPUT_LOCATIONS.veteran[:gender]
             },
-            'lastTreatmentFacility' => {
```
Comment (coope93) [inline lib/pdf_fill/forms/va1010cg.rb:129]: This field is removed on the new form.

File: lib/pdf_fill/forms/va1010cg.rb [hunk 4]
```diff
@@ -149,16 +168,21 @@ class Va1010cg < FormBase
               'street' => {
                 key: PDF_INPUT_LOCATIONS.primaryCaregiver[:address][:street],
                 limit: 80,
-                question_num: 25,
+                question_num: 208,
                 question_text: 'PRIMARY FAMILY CAREGIVER > Address > Street'
               }
             },
+            'mailingAddress' => {
```
Comment (coope93) [inline lib/pdf_fill/forms/va1010cg.rb:175]: Mailing Address is a newly added section to the 10-10CG

File: lib/pdf_fill/forms/va1010cg.rb [hunk 5]
```diff
@@ -149,16 +168,21 @@ class Va1010cg < FormBase
...
+                key: PDF_INPUT_LOCATIONS.primaryCaregiver[:mailingAddress][:street],
+                limit: 80,
+                question_num: 213,
+                question_text: 'PRIMARY FAMILY CAREGIVER > Mailing Address > Street'
+              }
+            },
             'gender' => {
               key: PDF_INPUT_LOCATIONS.primaryCaregiver[:gender]
             },
-            'hasHealthInsurance' => {
```
Comment (coope93) [inline lib/pdf_fill/forms/va1010cg.rb:159]: This was removed in the newest version of the 10-10CG.

File: lib/pdf_fill/forms/va1010cg.rb [hunk 6]
```diff
@@ -174,10 +198,18 @@ class Va1010cg < FormBase
               'street' => {
                 key: PDF_INPUT_LOCATIONS.secondaryCaregiverOne[:address][:street],
                 limit: 80,
-                question_num: 48,
+                question_num: 308,
```
Comment (coope93) [inline lib/pdf_fill/forms/va1010cg.rb:201]: These question numbers are rendered on an overflow page that is added for any form fields that do not fit in the pd form field. The old version was ordered 1,2,3 etc, while the new version has four different sections that all start at 1. To keep these changes as minimal as possible, the overflow page has them numbered with the section and then corresponding number. So 308 is section 3 question 08.

File: spec/fixtures/pdf_fill/10-10CG/kitchen_sink.json [hunk 7]
```diff
@@ -1,9 +1,10 @@
 {
   "veteran": {
       "fullName": {
-        "first": "Rosamond",
```
Comment (coope93) [inline spec/fixtures/pdf_fill/10-10CG/kitchen_sink.json:4]: This file was updated to match the new 1010CG schema.

File: spec/fixtures/pdf_fill/10-10CG/signed/kitchen_sink.pdf
Comment (coope93) [inline spec/fixtures/pdf_fill/10-10CG/signed/kitchen_sink.pdf:1]: These fixtures are filled out pdfs that the specs validate against. Kitchen sink has all of the values filled out, overflow has extra long entries with an `overflow` page, and simple is just the required fields.

**Review by JoshingYou1 — APPROVED** [review]
LGTM!

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**